src/sire/restraints/_restraints.py

```diff
@@ -656,6 +665,214 @@ def distance(mols, atoms0, atoms1, r0=None, k=None, name=None, map=None):
+        of the bond in the unperturbed state. Note that 'de' must still be provided.
+        Default is False.
+
+    Returns
+    -------
+    MorsePotentialRestraints : SireMM::MorsePotentialRestraints
+        A container of Morse restraints, where the first restraint is
+        the MorsePotentialRestraint created. The Morse restraint created can be
+        extracted with MorsePotentialRestraints[0].
+    """
+
+    from .. import u
+    from ..base import create_map
+    from ..mm import MorsePotentialRestraint, MorsePotentialRestraints
+    from ..morph import link_to_reference
+
+    map = create_map(map)
+
+    if k is None:
+        if not auto_parametrise:
+            raise ValueError("k must be provided if auto_parametrise is False")
+
+    elif type(k) is list:
+        k = [u(x) for x in k]
+    else:
+        k = [u(k)]
+    if auto_parametrise:
+        mol = mols.molecules("molecule property is_perturbable")
+        ref_mol = link_to_reference(mol)
+
+        if len(ref_mol) != 1:
+            raise ValueError(
+                "We need exactly one molecule that is perturbable to automatically set up the Morse potential restraints"
+            )
+        perturbable_mol = ref_mol[0]
+        pert = perturbable_mol.perturbation(map=map)
+        pert_omm = pert.to_openmm()
+        changed_bonds = pert_omm.changed_bonds(to_pandas=False)
+
+        # Attempt to find the bond that is annihilated at lambda=1
+        for bond in changed_bonds:
+            bond_name, length0, length1, k0, k1 = bond
+            if k1 == 0:
+
+                atom0_idx = [bond_name.atom0().index().value()][0]
+                atom1_idx = [bond_name.atom1().index().value()][0]
+
+                length0 = u(f"{length0} nm")
+
+                # Divide k0 by 2 to convert from force constant to sire half force constant k
+                if k is None:
+                    k0 = k0 / 2.0
+                    k0 = u(f"{k0} kJ mol-1 nm-2")
+                    k = [k0]
+
+                # Translate the atom numbers to the original system indexes
+                atoms0 = mols[
+                    f"molecule property is_perturbable and atomidx {atom0_idx}"
+                ]
+                atoms1 = mols[
+                    f"molecule property is_perturbable and atomidx {atom1_idx}"
+                ]
+                break
+
+    atoms0 = _to_atoms(mols, atoms0)
+    atoms1 = _to_atoms(mols, atoms1)
+
+    if atoms0.is_empty() or atoms1.is_empty():
+        raise ValueError("We need at least one atom in each group")
+
+    if len(atoms0) != len(atoms1):
+        raise ValueError("atoms0 and atoms1 must be the same length")
+
+    if len(atoms0) > 1 or len(atoms1) > 1:
+        if not auto_parametrise:
+            raise ValueError(
+                "Setting up multiplorse potential restraints at once is not currently supported. Please set up each restraint individually and then combine them into multiple restraints."
```

Comment:
Is `multiplorse` meant to say `mutiple Morse`? Although I like it, it could be a bit confusing. Also could the long line be broken into two similarly sized strings, e.g.

```python
           raise ValueError(
                "Setting up multiplorse potential restraints at once is not currently supported. "
                "Please set up each restraint individually and then combine them into multiple restraints."
           )
```
```